src/Butler/App/TodoManager.hs

```diff
@@ -321,16 +408,48 @@ startTodoManager ctx = do
                              ) of
                             (Just taskID, Just taskDesc, Just taskPrio, Just taskDueDate') -> do
-                                taskDueDate <- parseTimeM False defaultTimeLocale defaultDateFormat taskDueDate'
+                                taskDueDate <- parseDueDate taskDueDate'
                                 atomically $ modifyMemoryVar appStateM $ \tm -> do
-                                    updateTask (unSetEditingTask tm) (TaskId taskID) (TaskDesc taskDesc) (textToPrio taskPrio) (TaskDueDate taskDueDate)
+                                    updateTask (unSetEditingTask tm) (TaskId taskID) (TaskDesc taskDesc) (textToPrio taskPrio) taskDueDate
                             _ -> pure ()
                     "checkbox-click" -> do
                         logInfo "Selected item" ["ev" .= ev]
                         case ev.body ^? key "taskID" . _Integer of
                             Just taskId -> do
                                 atomically $ modifyMemoryVar appStateM $ \tm -> do
-                                    setSelectedTask tm $ TaskId $ fromInteger $ toInteger taskId
+                                    setSelectedTask (TaskId $ fromInteger $ toInteger taskId) tm
                             Nothing -> pure ()
+                    "show-settings" -> do
+                        logInfo "Show settings" ["ev" .= ev]
+                        atomically $ modifyMemoryVar appStateM $ \tm -> do
+                            tm{todoSettingsShow = not tm.todoSettingsShow}
+                    "setting-show-column-prio" -> do
+                        logInfo "Show settings" ["ev" .= ev]
+                        atomically $ modifyMemoryVar appStateM $ \tm -> do
+                            tm
+                                { todoSettings =
+                                    tm.todoSettings
+                                        { showColumnPrio = not tm.todoSettings.showColumnPrio
+                                        }
+                                }
+                    "setting-show-column-dueDate" -> do
+                        logInfo "Show settings" ["ev" .= ev]
+                        atomically $ modifyMemoryVar appStateM $ \tm -> do
+                            tm
+                                { todoSettings =
+                                    tm.todoSettings
+                                        { showColumnDueDate = not tm.todoSettings.showColumnDueDate
+                                        }
+                                }
                     _ -> logError "Unknown trigger" ["ev" .= ev]
             ev -> logError "Unknown ev" ["ev" .= ev]
+
+parseDueDate :: String -> ProcessIO TaskDueDate
+parseDueDate taskDueDate = do
+    res <- runExceptT $ do
+        if null taskDueDate
+            then pure NoDueDate
+            else DueDate <$> parseTimeM False defaultTimeLocale defaultDateFormat taskDueDate
```

Comment:
nit: since you go for ExceptT, that should work too:

```suggestion
        when (null taskDueDate) do
          throwError "Empty date"
        DueDate <$> parseTimeM False defaultTimeLocale defaultDateFormat taskDueDate
```